src/Bijectors.jl

```diff
@@ -67,13 +67,23 @@ function maphcat(f, args...)
     init = reshape(out[1], :, 1)
     return reduce(hcat, drop(out, 1); init = init)
 end
-function eachcolmapvcat(f, x)
-    return mapvcat(f, eachcol(x))
+function eachcolmaphcat(f, x1, x2)
+    out = [f(x1[:,i], x2[i]) for i in 1:size(x1, 2)]
+    init = reshape(out[1], :, 1)
+    return reduce(hcat, drop(out, 1); init = init)
+end
+function eachcolmaphcat(f, x)
+    out = map(f, eachcol(x))
+    init = reshape(out[1], :, 1)
+    return reduce(hcat, drop(out, 1); init = init)
+end
```

Comment:
What is the reason for defining `mapvcat`, `maphcat`, `eachcolmaphcat`, and `eachcolmapvcat` instead of, e.g., using `mapreduce(hcat, f, eachcol(x))` for `eachcolmaphcat(f, x)`? Type instabilities? Lately I managed to solve some type stability issues with `mapreduce` by using `reshape(x, Val(...))`, IIRC - maybe something similar could be applicable here, depending on your use case?

Reply:
No type stability is not an issue. It's just that Zygote doesn't like `eachcol` or `mapreduce`, they error. So I define these functions to be used by Tracker then define their adjoint using something that Zygote can differentiate.

Reply:
Ah I see :slightly_smiling_face: